src/repositories/docsetsRepository.ts

```diff
@@ -9,17 +9,19 @@ export class DocsetsRepository extends BaseRepository {
     super(config, logger, 'DocsetsRepository', db.collection(docsetsCollectionName));
   }
 
-  private getAggregationPipeline(
-    matchConditionField: string,
-    matchConditionValue: string,
-    projection?: { [k: string]: number }
-  ) {
+  static getAggregationPipeline(matchConditions: { [k: string]: string }, projection?: { [k: string]: number }) {
```

Comment:
NIT: It would be nice to provide a more concrete type for `matchConditions` and `projection` so that if we use different values for `matchConditions` or `projection` we know what we can use.

[Here's a link to how this could be done](https://www.typescriptlang.org/play?#code/C4TwDgpgBAIglgJwgY2AewXCBnA0hEKAXigCJsA7NNUAfXQFsAbUgKFEigCUIw0AhBAEMKyABY58hEqTAI0AKxTBSUAD5kARsNETsqjaSR8AckIYQDUAAYATRMoxZsAOgAkAb3hJUTyQQBfa1ZWAHpQqA4cZEwwYCgAdzgmJigAayoEyLEheOAxOGwoADchJgBXaEKoIWRkXmAhTSZoADMMbOgePkERcX9CKNZkNApseJGKWwBxalsALm5eAR1+vAJiMnsfdEwcF0pqOkYWEPDI8GjY+IgEeQQiiSQoTRQhcuxoAHJtxz3XTRHED0NDML5QapUeJCEplOC2WEVNodborPp6KTDUbjKCTWz8IQLJY9VYYjYyX6+f4uQE0YEnNhhCIJaDIERQD7QfLQBhCMCQBFRKDYNDZXKJaC2UZfeIUCAQQWipiFG7FW6EbBwCgAcxaiMqUFa8gYxLRugGTOFWvqNRSnQukCKQme40wOqdFBACSEIFcUAAgrZBRIoF8ANQAfnBmtE0ChUAgaoQhDSGySdteUFsEFaWoVLnYlygAFlcuIAMKjezAOCjTYeKAAbVThC1pt65vWIAAupHFq6tdqoAEQpMcbzgBWq3Aa9jFqXJ2JK1MZ7WKPXWFAt6HKbtnAcqHSQWDFl9kEw0OVbABaKXIbChQ50lwnL6sEdAA)

Reply:
ahhh, good old `booty_toml`